File: 2026/array/maximum_matrix_sum_1975.cpp
// Meta

// Check Odd or Even count of -ves
// T.C : O(row*col)
// S.C : O(1)

class Solution {
public:
    long long maxMatrixSum(vector<vector<int>>& matrix) {
        int n = matrix.size();

        long long sum = 0;
        int countNegatives = 0;
        int smallestAbsoluteValue = INT_MAX;

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                sum += abs(matrix[i][j]);

                countNegatives += matrix[i][j] < 0 ? 1 : 0;

                smallestAbsoluteValue =
                    min(smallestAbsoluteValue, abs(matrix[i][j]));
            }
        }

        if (countNegatives % 2 == 0) {
            return sum;
        }

        return sum - 2 * smallestAbsoluteValue;
    }
};

File: 2026/array/maximum_matrix_sum_1975.py
class Solution:
    def maxMatrixSum(self, matrix: List[List[int]]) -> int:
        n = len(matrix)
        total_sum = 0
        count_negatives = 0
        smallest_abs_value = float("inf")
        for i in range(n):
            for j in range(n):
                val = matrix[i][j]
                total_sum += abs(val)
                if val < 0:
                    count_negatives += 1
                smallest_abs_value = min(smallest_abs_value, abs(val))

        if count_negatives % 2 == 0:
            return total_sum
        
        return total_sum - 2 * smallest_abs_value
